includes/User.php

```diff
@@ -3124,15 +3133,25 @@ public function isAnon() {
 	}
 
 	/**
-	 * Get whether the user is the current performer
+	 * Get whether the instance represents the current performer
 	 *
-	 * Inspired by User::equals() and its usage in MW 1.25
 	 * @return Bool
 	 * @author Michał Roszka <michal@wikia-inc.com>
 	 */
 	public function isCurrent() {
+		// This is the first implementation. I still rely on $wgUser here
+		// and want to log all cases when mIsCurrent property is false
+		// for the current performer and fix it for future checks.
 		global $wgUser;
-		return $this->getName() === $wgUser->getName();
+		if ( !$this->mIsCurrent && $this->getName() === $wgUser->getName() ) {
```

Comment:
Isn't the user's id the only invariant that we have for the user? Should we be comparing user id instead of the name?